### PR TITLE
Refactor Smart Fridges, allow auto-menders to be put in Medical Fridges

### PR DESCRIPTION
**Pull request by dearmochi**
## What Does This PR Do

- **Completely refactors, documents and cleans Smart Fridge code.**
- Auto-menders are now accepted by Refrigerated Medicine Storages.
- Fixes an exploit with `preloaded` Smart Fridges where deconstructing them then rebuilding them would generate a new set of items.

## Why It's Good For The Game

- (Hopefully) higher code quality.
- Pill bottles, patch packs are accepted by the Medical Fridge. Auto-menders should be accepted too - eliminates the need of putting them on the floor/a table and generally helps keep things organized.
- One less duplication exploit.

## Changelog
:cl:
tweak: Medical Fridges now accept auto-menders
fix: Fix a duplication exploit with preloaded Smart Fridges
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 1]
```diff
@@ -79,6 +79,8 @@
 		return 1
 	if(istype(O,/obj/item/reagent_containers/iv_bag))
 		return 1
+	if(istype(O,/obj/item/reagent_containers/applicator))
```
Comment (farie82) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:82]: Spacing

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 2]
```diff
@@ -79,6 +79,8 @@
...
+		return 1
```
Comment (farie82) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:83]: `TRUE` here

**Comment by dearmochi** [comment]
So I went and refactored and documented the entire file, which is why I took so long updating this.

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 3]
```diff
@@ -429,36 +281,342 @@
...
+  * Secure variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  * Can be emagged and EMP'd to short the lock.
+  */
+/obj/machinery/smartfridge/secure
+	is_secure = TRUE
+
+/obj/machinery/smartfridge/secure/emag_act(mob/user)
+	emagged = TRUE
+	locked = SMART_FRIDGE_LOCK_SHORTED
+	to_chat(user, "You short out the product lock on [src].")
```
Comment (SteelSlayer) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:367]: Needs spans.

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 4]
```diff
@@ -429,36 +281,342 @@
...
+  * # MegaSeed Servitor
+  *
+  * Seeds variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  */
+/obj/machinery/smartfridge/seeds
+	name = "\improper MegaSeed Servitor"
+	desc = "When you need seeds fast!"
+	icon = 'icons/obj/vending.dmi'
+	icon_state = "seeds"
+
```
Comment (AffectedArc07) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:397]: This is the exact same name as the vending machine. This is not only confusing, but made me double check that botany actually used a proper vending machine.

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 5]
```diff
@@ -429,36 +281,342 @@
...
+/**
+  * # Slime Extract Storage
+  *
+  * Secure, Xenobiology variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  */
+/obj/machinery/smartfridge/secure/extract
+	name = "\improper Slime Extract Storage"
+	desc = "A refrigerated storage unit for slime extracts"
+	req_access_txt = "47"
```
Comment (AffectedArc07) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:432]: Any reason to not use `req_access_txt = "[ACCESS_RESEARCH]"`?

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 6]
```diff
@@ -429,36 +281,342 @@
...
+/**
+  * # Secure Refrigerated Medicine Storage
+  *
+  * Secure, Medical variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  */
+/obj/machinery/smartfridge/secure/medbay
+	name = "\improper Secure Refrigerated Medicine Storage"
+	desc = "A refrigerated storage unit for storing medicine and chemicals."
+	icon_state = "smartfridge" //To fix the icon in the map editor.
+	req_one_access_txt = "5;33"
```
Comment (AffectedArc07) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:449]: Same here

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 7]
```diff
@@ -429,36 +281,342 @@
...
+/**
+  * # Smart Chemical Storage
+  *
+  * Secure, Chemistry variant of the [Smart Fridge][/obj/machinery/smartfridge].
+  */
+/obj/machinery/smartfridge/secure/chemistry
+	name = "\improper Smart Chemical Storage"
+	desc = "A refrigerated storage unit for medicine and chemical storage."
+	icon_state = "smartfridge" //To fix the icon in the map editor.
+	req_access_txt = "33"
```
Comment (AffectedArc07) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:470]: And here

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm (hunk 4 above)
Comment (dearmochi) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: Wasn't aware of that, I renamed it to `Seed Storage` (bland, I know)

**Pull request merged**